app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ImportExportApplicationService.java

```diff
@@ -0,0 +1,576 @@
+                            }
+                            if (newAction.getUnpublishedAction() != null) {
+                                ActionDTO actionDTO = newAction.getUnpublishedAction();
+                                actionDTO.setPageId(newPageIdMap.get(actionDTO.getPageId()));
+                            }
+                        });
+                        applicationJson
+                                .getDatasourceList()
+                                .removeIf(datasource -> !concernedDBNames.contains(datasource.getName()));
+
+                        return newActionList;
+                    })
+                    .map(newActionList -> {
+                        applicationJson.setActionList(newActionList);
+                        //Only export those datasources which are used in the app instead of org level
+                        Map<String, DecryptedSensitiveFields> decryptedFields = new HashMap<>();
+                        applicationJson.getDatasourceList().forEach(datasource -> {
+                            decryptedFields.put(datasource.getName(), getDecryptedFields(datasource));
+                            datasource.setId(null);
+                            datasource.setOrganizationId(null);
+                            datasource.setPluginId(pluginMap.get(datasource.getPluginId()));
+                            if (datasource.getDatasourceConfiguration() != null) {
+                                datasource.getDatasourceConfiguration().setAuthentication(null);
+                            }
+                        });
+                        applicationJson.setDecryptedFields(decryptedFields);
+                        return Mono.just(applicationJson);
+                    });
+                })
+                .then()
+                .thenReturn(applicationJson);
+    }
+
+    public Mono<Application> extractFileAndSaveApplication(String orgId, Part filePart) {
+
+        final MediaType contentType = filePart.headers().getContentType();
+        
+        if (orgId == null || orgId.isEmpty()) {
+            return Mono.error(new AppsmithException(AppsmithError.INVALID_PARAMETER, FieldName.ORGANIZATION_ID));
+        }
+        
+        if (contentType == null || !ALLOWED_CONTENT_TYPES.contains(contentType)) {
+            return Mono.error(new AppsmithException(
+                AppsmithError.VALIDATION_FAILURE,
+                FieldName.INVALIDJSONFILE
+            ));
+        }
+
+        final Flux<DataBuffer> contentCache = filePart.content().cache();
+        Mono<String> stringifiedFile = contentCache.count()
+            .defaultIfEmpty(0L)
+            .flatMap(count -> {
+                // Default implementation for the BufferFactory used breaks down the FilePart into chunks of 4KB.
+                // So we multiply the count of chunks with 4 to get an estimate on the file size in KB.
+                return DataBufferUtils.join(contentCache);
+            })
```

Comment:
It looks like we are not using the `count` value anywhere right? Can we just get the content directly?

```suggestion
        Mono<String> stringifiedFile = DataBufferUtils.join(contentCache)
```

Reply:
Updated code pushed!